llama-util.h

```diff
@@ -550,4 +553,107 @@ struct llama_ctx_buffer {
 typedef llama_buffer llama_ctx_buffer;
 #endif
 
+// returns true if successful, false otherwise
+static bool create_directory_with_parents(const std::string & path) {
```

Comment:
I would advice to just avoid creating directories from C++.

The calling script / program passes the path to the directory that you want to create, so it is trivial for it to create the directory before calling `llama.cpp`. Will save you the hurdle of dealing with filesystem calls.

Reply:
What exactly are you suggesting? Calling e.g. `mkdir -P` on Unix-like systems? Because that would open the door to put arbitrary commands in the `--logdir` CLI argument.

Reply:
My suggestion is to simply create the path before running the perplexity tool.
Instead of:

```bash
./perplexity --logdir /some/path ...
```

do

```bash
mkdir -p /some/path
./perplexity --logdir /some/path ...
```

And in the code you assume that the path is guaranteed to be created, or throw error if the path does not exist.
No command execution from C++ and no filesystem calls are needed this way

Reply:
Ok lets keep it as it is. We can reconsider later if for some reason this starts causing some cross-compatibility troubles